python/django/security/user-open.yaml

```diff
@@ -0,0 +1,66 @@
+rules:
+  - id: user-open
+    patterns:
+      - pattern-inside: |
+          def $F(...):
+            ...
+      - pattern-either:
+          - pattern: open(..., request.$W.get(...), ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              open(..., $V, ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              $A = open(..., $V, ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              return open(..., $V, ...)
+          - pattern: open(..., request.$W(...), ...)
+          - pattern: |
+              $V = request.$W(...)
+              ...
+              open(..., $V, ...)
+          - pattern: |
+              $V = request.$W(...)
+              ...
+              $A = open(..., $V, ...)
+          - pattern: |
+              $V = request.$W(...)
+              ...
+              return open(..., $V, ...)
+          - pattern: open(..., request.$W[...], ...)
+          - pattern: |
+              $V = request.$W[...]
+              ...
+              open(..., $V, ...)
+          - pattern: |
+              $V = request.$W[...]
+              ...
+              $A = open(..., $V, ...)
+          - pattern: |
+              $V = request.$W[...]
+              ...
+              return open(..., $V, ...)
+          - pattern: |
+              $V = request.$W.get(...)
+              ...
+              with open($V, ...) as $FD:
+                ...
+          - pattern: |
+              $V = request.$W(...)
+              ...
+              with open($V, ...) as $FD:
+                ...
+          - pattern: |
+              $V = request.$W[...]
+              ...
+              with open($V, ...) as $FD:
+                ...
+    message: |
```

Comment:
Slight nit here: I would change `Found user data ...` -> `Found request data ...` and `Make absolutely sure ...` -> `Please ensure ...`.